design-documents/checkout/alternative-checkout-flow.md

```diff
@@ -0,0 +1,63 @@
+# Alternative checkout flow
+
+In scope of work on GraphQL and storefront APIs we have an opportunity to improve design and features of storefront checkout.
+
+The purpose of this document is to discuss possible alternatives to current Magento checkout flow. 
+
+The cart is just a container for the items the user wants to purchase. In the proposed flow the cart is created as soon as user adds product to cart. The data from cart entity should be enough to render minicart and cart pages. Taxes and other adjustments are not calculated at these steps.
+
+The quote, on the other hand, contains a full break down of all adjustments calculated. It provides the user with the total he has to pay for the items in the cart.
+
+It is possible to split cart items into separate quotes. This can be done based on shipping addresses or shipping sources.
+
+Each quote is used to create a separate order. Multiple payment methods (with independent billing address each) can be selected during each order creation.
+
+Cart will depend on Catalog. Quote will depend on Cart and must not depend on Catalog. Order will depend on Quote and must not depend on Cart or Catalog.
```

Comment:
> Quote will depend on Cart and must not depend on Catalog

Why?
I don't see the benefit of this. I see the attraction of it, but I think it's wrong. It's an unnatural restriction. Signs of this are how awkward it is to pass the product dimensions. The same problems will occur also for example when other product data is required for totals calculation by third parties.